xonsh/built_ins.py

```diff
@@ -638,7 +638,7 @@ def load(self, execer=None, ctx=None, inherit_env=True, **kwargs):
         else:
             self.env = Env({"XONSH_ENV_INHERITED": False})
 
-        self.exit = False
+        self.exit = None
```

Comment:
Looks good to me. I think I'd like to see some comments or docstring that explains what `.exit` is, what values it might be, and what those values mean (mainly `None` means not exiting, but an integer means it is exiting and with that code).